### PR TITLE
Bug/9548 Fix and refactor view only KMW tiles

### PR DESCRIPTION
**Pull request by jimmymadon**
## Summary

Addresses issue:

- #9548 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.
- [x] Ensure there are no unexpected significant changes to file sizes.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Review by benbowler — APPROVED** [review]
Reviewed, tested and addressed merge conflicts, moving to MR.

**Review by benbowler — CHANGES_REQUESTED** [review]
After rechecking the VRTs they are in fact failing now with the button missing in the `KeyMetrics/ChangeMetricsLink` story.

**Comment by jimmymadon** [comment]
@benbowler Thanks for that - I tried running the VRTs 3 times and I know you ran them for 3 times as well! This is now fixed.

File: assets/js/googlesitekit/datastore/user/key-metrics.js [hunk 1]
```diff
@@ -411,7 +392,46 @@ const baseSelectors = {
...
+
+				if ( ! widget ) {
+					return false;
+				}
+
+				if (
+					widget.displayInWidgetArea &&
+					typeof widget.displayInWidgetArea === 'function'
+				) {
+					return widget.displayInWidgetArea(
```
Comment (aaemnnosttv) [inline assets/js/googlesitekit/datastore/user/key-metrics.js:416]: I see this was something we were doing before as well but it seems incorrect to make this selector view-only aware and specific to displaying in the widget area which seem like concerns that are more localized to the React side of things. I guess we can keep it here for simplicity regarding this PR but it does seem like there is room for decoupling these things a bit more.

**Review by aaemnnosttv — APPROVED** [review]
Thanks @jimmymadon – this looks like it addresses the potential infinite recursion and doesn't error in my testing. A few other comments for your consideration but we can consider these for a follow up issue(s).

File: assets/js/googlesitekit/datastore/user/key-metrics.js [hunk 2]
```diff
@@ -224,12 +223,6 @@ const baseSelectors = {
 		}
 
 		if ( userPickedMetrics.length ) {
-			if ( ! isFeatureEnabled( 'conversionReporting' ) ) {
-				return userPickedMetrics.filter( ( slug ) => {
-					return keyMetricsGA4WidgetsNonACR.includes( slug );
-				} );
-			}
```
Comment (zutigrm) [inline assets/js/googlesitekit/datastore/user/key-metrics.js:231]: This seems to be deleted by mistake, this is used in ACR epic, and is important to clear the metrics when feature flag is disabled, I will restore it in my current PR

File: assets/js/googlesitekit/datastore/user/key-metrics.js [hunk 3]
```diff
@@ -240,12 +233,6 @@ const baseSelectors = {
 		}
 
 		if ( answerBasedMetrics.length ) {
-			if ( ! isFeatureEnabled( 'conversionReporting' ) ) {
-				return answerBasedMetrics.filter( ( slug ) => {
-					return keyMetricsGA4WidgetsNonACR.includes( slug );
-				} );
-			}
-
```
Comment (zutigrm) [inline assets/js/googlesitekit/datastore/user/key-metrics.js:248]: Same here, will restore it in my current PR

File: assets/js/googlesitekit/datastore/user/key-metrics.js (hunk 2 above)
Comment (jimmymadon) [reply assets/js/googlesitekit/datastore/user/key-metrics.js:231]: Here we've moved the filtering to within `getUserPickedMetrics` along with a new `displayInWidgetArea` check.

https://github.com/google/site-kit-wp/blob/f625d0286075066553879b4f26f667b20c032d78/assets/js/googlesitekit/datastore/user/key-metrics.js#L400-L425

File: assets/js/googlesitekit/datastore/user/key-metrics.js (hunk 3 above)
Comment (jimmymadon) [reply assets/js/googlesitekit/datastore/user/key-metrics.js:248]: @zutigrm  In `getAnswerBasedMetrics`, we already check for the feature flag and adjust the values of what is returned. So no point in doing this again.

**Comment by jimmymadon** [comment]
@aaemnnosttv Thanks for the MR here - I've created a new issue #9645 to address your PR feedback from this issue.

**Pull request merged**